external/radiation/python/radsw/radsw_main.py

```diff
@@ -3141,962 +3363,1051 @@ def taumol20(
+                " radiation cloud radiative property setup !!",
+            )
+
+        if self.isubcsw == 0 and self.iovrsw > 2:
+            if me == 0:
+                warnings.warn(
+                    f"*** IOVRSW={self.iovrsw} is not available for",
+                    " ISUBCSW=0 setting!!",
                 )
+                warnings.warn(
+                    "The program will use maximum/random overlap", " instead."
+                )
+            self.iovrsw = 1
 
-        return taug, taur
+        #  --- ...  setup constant factors for heating rate
+        #           the 1.0e-2 is to convert pressure from mb to N/m**2
 
-    # The subroutine computes the optical depth in band 22:  7700-8050
-    # cm-1 (low - h2o,o2; high - o2)
+        if iswrate == 1:
+            self.heatfac = con_g * 864.0 / con_cp  #   (in k/day)
+        else:
+            self.heatfac = con_g * 1.0e-2 / con_cp  #   (in k/second)
 
-    def taumol22(
-        self,
-        colamt,
-        colmol,
-        fac00,
-        fac01,
-        fac10,
-        fac11,
-        jp,
-        jt,
-        jt1,
-        laytrop,
-        forfac,
-        forfrac,
-        indfor,
-        selffac,
-        selffrac,
-        indself,
-        nlay,
-        id0,
-        id1,
-        taug,
-        taur,
-        ds,
-    ):
+        #  --- ...  define exponential lookup tables for transmittance. tau is
+        #           computed as a function of the tau transition function, and
+        #           transmittance is calculated as a function of tau.  all tables
+        #           are computed at intervals of 0.0001.  the inverse of the
+        #           constant used in the Pade approximation to the tau transition
+        #           function is set to bpade.
 
-        #  ------------------------------------------------------------------  !
-        #     band 22:  7700-8050 cm-1 (low - h2o,o2; high - o2)               !
-        #  ------------------------------------------------------------------  !
-        #
+        self.exp_tbl = np.zeros(ntbmx + 1)
+        self.exp_tbl[0] = 1.0
+        self.exp_tbl[ntbmx] = expeps
 
-        selfref = ds["selfref"].values
-        forref = ds["forref"].values
-        absa = ds["absa"].values
-        absb = ds["absb"].values
-        rayl = ds["rayl"].values
+        for i in range(ntbmx - 1):
+            tfn = i / (ntbmx - i)
+            tau = self.bpade * tfn
+            self.exp_tbl[i] = np.exp(-tau)
 
-        #  --- ...  the following factor is the ratio of total o2 band intensity (lines
-        #           and mate continuum) to o2 band intensity (line only). it is needed
-        #           to adjust the optical depths since the k's include only lines.
+    def return_initdata(self):
+        outdict = {"heatfac": self.heatfac, "exp_tbl": self.exp_tbl}
+        return outdict
 
-        o2adj = 1.6
-        o2tem = 4.35e-4 / (350.0 * 2.0)
+    def swrad(
```

Comment:
I expect you'll see a speedup comparable to lwrad improvement if you jit this.

Reply:
Next PR though?

Reply:
I agree. I can try to speedup this part in another PR.